src/tests/unit-tests/parse_service.ml

```diff
@@ -0,0 +1,109 @@
+module E = Pb_exception
+module Pt = Pb_parsing_parse_tree 
+
+let parse f s  = 
+  f Pb_parsing_lexer.lexer (Lexing.from_string s)
+
+let expect asserted expected = 
```

Comment:
have you considered using a testing tool like https://github.com/mirage/alcotest for the unit tests?

I haven't looked into alcotest, but have found https://reason-native.com/docs/rely/ (requires esy) snapshot tests to be very helpful for testing code generation since the output is automatically generated and then tested for regressions

Reply:
Would love to use a minimal dependency testing framework. Something to look into.